test/run/variants.mo

```diff
@@ -55,3 +55,7 @@ assert (debug_show (#foo (#bar)) == "#foo(#bar)");
 
 assert ([#Monday, #Tuesday, #Wednesday, #Thursday, #Friday, #Saturday, #Sunday].size() == 7);
 
+// check compilation of pattern matching on constant variants
+let (#const a) = #const 42;
```

Comment:
Also a check for a failed constant match?

Reply:
I can add a test. But the new predicate should fail there too, and as such the old path will be taken.

Reply:
You are probably right, and the exceptions will be caught in the new predicate, but that also means that the runtime pattern match will be generated with the same problems :-(

To make this bottom-aware we need a three-way predicate.